pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/SchemaDefinitionBuilderImpl.java

```diff
@@ -114,6 +120,18 @@
         return this;
     }
 
+    @Override
+    public SchemaDefinitionBuilder<T> withSchemaReader(SchemaReader<T> reader) {
+        this.reader=reader;
+        return null;
```

Comment:
I think you meant `return this;`, which begs the question, can we have some tests?

Reply:
Welll ... obviously I meant that ... you should never do two things at once ...